# **Koby's CS111 Notebook**
My hope is to get some fairly accurate models to put together to an ensamble.




In [9]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.neural_network import MLPClassifier
from sklearn.neural_network import MLPRegressor

from sklearn.neighbors import KNeighborsClassifier
from sklearn.neighbors import KNeighborsRegressor

from sklearn.tree import DecisionTreeRegressor
from sklearn.model_selection import train_test_split

In [5]:
# Fetch training data
#!curl --remote-name \-H 'Accept: application/vnd.github.v3.raw' \--location https://github.com/petridish123/CS201R_CS111/blob/89c6e77108581e0ecc1f0eaf7ed57f9a91b3f9fe/Koby/pre_mt2_train.csv
#file doesnt work when i use that ^^

#Load and split data
grades_upto_mid2 = pd.read_csv("pre_mt2_train.csv")
X = grades_upto_mid2.iloc[:,:-2]
y_scores = grades_upto_mid2.iloc[:,-2]

y_grades = grades_upto_mid2.iloc[:,-1]
#print(y_scores)
X_train, X_test, y_train, y_test = train_test_split(X,y_grades, test_size=.2)
X_train_scores, X_test_scores, y_train_scores, y_test_scores, = train_test_split(X,y_scores, test_size=.2)


In [6]:
mlp = MLPClassifier(hidden_layer_sizes=(16,4,2,1), activation='logistic', solver= 'sgd', alpha= 0.01, batch_size= 1, learning_rate_init= .01, shuffle= True, momentum= 0, n_iter_no_change=50, max_iter=1000,)
mlp.fit(X_train, y_train)
print("Training set score: %f" % mlp.score(X_train, y_train))
print("Test set score: %f" % mlp.score(X_test, y_test))

Training set score: 0.305556
Test set score: 0.353659


In [7]:
mlp = MLPRegressor(hidden_layer_sizes=(16,), activation='logistic', solver= 'sgd', alpha= 0, batch_size= 1, learning_rate_init= .001, shuffle= True, momentum= 0, n_iter_no_change=50, max_iter=1000,)
mlp.fit(X_train_scores, y_train_scores)
print("Training set score: %f" % mlp.score(X_train_scores, y_train_scores))
print("Test set score: %f" % mlp.score(X_test_scores, y_test_scores))

Training set score: 0.431713
Test set score: 0.369564


# **MLP Classifier**

With starting with 30 ish% testing accuracy.
I tried a whole bunch of different layers and whatnot and couldnt get anything better.
# **MLP Regressor**

About the same as the classifier with basic parameters with a learning rate of .001 and 16, hidden layers i got 37%

In [8]:
clf = KNeighborsClassifier(n_neighbors= 10, weights= 'distance')
clf.fit(X_train,y_train)
print(f"Training score: {clf.score(X_train, y_train)}")
print(f"Testing score: {clf.score(X_test, y_test)}")

Training score: 0.8148148148148148
Testing score: 0.35365853658536583


In [10]:
clf = KNeighborsRegressor(n_neighbors= 10, weights= 'distance')
clf.fit(X_train_scores,y_train_scores)
print(f"Training score: {clf.score(X_train_scores, y_train_scores)}")
print(f"Testing score: {clf.score(X_test_scores, y_test_scores)}")

Training score: 0.9728933318181986
Testing score: 0.6488728168403193


# **KNN Classifier**

Right away the test score is low, around 20%.
# **KNN Regressor**
Was able to get 65%